xbmc/pvr/providers/PVRProviders.cpp

```diff
@@ -105,6 +105,8 @@ bool CPVRProviders::LoadFromDatabase(const std::vector<std::shared_ptr<CPVRClien
   const std::shared_ptr<CPVRDatabase> database = CServiceBroker::GetPVRManager().GetTVDatabase();
   if (database)
   {
+    m_iLastId = database->GetMaxProviderId();
```

Comment:
Still not very clean, as the value is still obtained multiple times from the db, where one time should be enough. But if you don't find a better solution, this will be okay I guess.

Reply:
I agree, it could be cleaner, but enough here for this fix.

The providers code will require some work to make it instance aware so improve it then.